cmd/serve_test.go

```
@@ -17,9 +28,142 @@ func TestServeCommandDefined(t *testing.T) {
+	}
+}
+
+// serveTestTimeout bounds every wait in the lifecycle test.
+const serveTestTimeout = 5 * time.Second
+
+// freePort asks the OS for a port and releases it, so runServe can bind it.
+// The gap between releasing and rebinding is a race in principle; in practice
+// this is the standard technique and collisions do not happen in test runs.
+func freePort(t *testing.T) int {
+	t.Helper()
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)